integritybackend/claim.py

```diff
@@ -97,19 +97,63 @@ def generate_create(self, jwt_payload, data):
 
-    def generate_create_proofmode(self, jwt_payload, meta_proofmode):
-        """Generates a claim for the 'create_proofmode' action.
+    def generate_c2pa_proofmode(self, action_params: dict, meta_content: dict, filename: str):
+        """Generates a claim for the 'c2pa-proofmode' action.
 
         Args:
-            jwt_payload: a dictionary with the data we got from the request's JWT payload
-            meta_proofmode: dictionary with the metadata from a proofmode bundle
+            action_params: a dictionary with the params from this action's config
+            meta_content: dictionary with the metadata from a proofmode bundle
+            filename: 
 
         Returns:
             a dictionary containing the 'create' claim data
         """
         claim = copy.deepcopy(CREATE_CLAIM_TEMPLATE)
+        claim["recorder"] = "ProofMode by Guardian Project and WITNESS"
+
+        assertion_templates = self.assertions_by_label(claim)
+        assertions = []
+
+        author_data = action_params["creative_work_author"]
+        if author_data is not None:
+            creative_work = assertion_templates["stds.schema-org.CreativeWork"]
+            creative_work["data"] = author_data
+            assertions.append(creative_work)
+
+        author_name = action_params["creative_work_author"]["name"]
+        copyright = action_params["copyright"]
+        gps_lat = meta_content["private"]["proofmode"][filename]["proofs"][0]["Location.Latitude"]
+        gps_lon = meta_content["private"]["proofmode"][filename]["proofs"][0]["Location.Longitude"]
+        photo_meta_data = self._make_photo_meta_data(author_name, copyright, gps_lat, gps_lon)
+        if photo_meta_data is not None:
+            photo_meta = assertion_templates["stds.iptc.photo-metadata"]
+            photo_meta["data"] = photo_meta_data
+            assertions.append(photo_meta)
+
+        gps_time = date.fromtimestamp(int(meta_content["private"]["proofmode"][filename]["proofs"][0]["Location.Time"])/1000).isoformat()
+        exif_data = self._make_exif_data(float(gps_lat), float(gps_lon), gps_time)
+        if exif_data is not None:
+            exif = assertion_templates["stds.exif"]
+            exif["data"] = exif_data
+            assertions.append(exif)
+
+        pgp_sig = meta_content["private"]["proofmode"][filename]["pgpSignature"]
+        pgp_pubkey = meta_content["private"]["proofmode"][filename]["pgpPublicKey"]
+        sha256hash = meta_content["private"]["proofmode"][filename]["sha256hash"]
+        signature_data = self._make_signature_data_proofmode(pgp_sig, pgp_pubkey, sha256hash, filename)
+        if signature_data is not None:
+            signature = assertion_templates["org.starlinglab.integrity"]
+            signature["data"] = signature_data
+            assertions.append(signature)
+
+        # proofmode_time = meta_content["dateCreated"]
+        proofmode_time = meta_content["private"]["proofmode"][filename]["proofs"][0]["Proof Generated"]
```

Comment:
@YurkoWasHere we need to change this to your new update. What should it be, can you suggest?

Reply:
`meta_content["dateCreated"]`
 or 
`meta_content["private"]["proofmode"]["dateCreate"]`
both are the same value, 

It is the date/time that a file INSIDE the zip was modified, therefore it is when the proof was zipped up by proof mode.